aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -888,6 +896,119 @@ module aptos_framework::delegation_pool {
+            stake_pool_signer_cap,
+            // `staking_contract::distribute_internal` has already withdrawn any existing inactive stake
+            total_coins_inactive: 0,
+            operator_commission_percentage,
+            add_stake_events: account::new_event_handle<AddStakeEvent>(&stake_pool_signer),
+            reactivate_stake_events: account::new_event_handle<ReactivateStakeEvent>(&stake_pool_signer),
+            unlock_stake_events: account::new_event_handle<UnlockStakeEvent>(&stake_pool_signer),
+            withdraw_stake_events: account::new_event_handle<WithdrawStakeEvent>(&stake_pool_signer),
+            distribute_commission_events: account::new_event_handle<DistributeCommissionEvent>(&stake_pool_signer),
+        };
+
+        // Grant ownership of the existing stake to the participants of the destroyed staking contract.
+        // 1. `synchronize_delegation_pool` MUST NOT be called during this process because stake would be distributed as
+        // rewards instead of being individually allocated
+        // 2. partial governance voting remains disabled for simplicity
+
+        // allocate active stake to staker, any active commission has already been unlocked + allocated to operator
+        // no need to assert on minimum required stake as any next operation of delegator enforces it
+        buy_in_active_shares(&mut pool, staker_address, principal);
+
+        // allocate pending_inactive stake to staker, current operator, any previous operators still pending_inactive
+        while (pool_u64_iterable::shareholders_count(&distribution_pool) > 0) {
+            let recipients = pool_u64_iterable::shareholders(&distribution_pool);
+            let recipient = *vector::borrow(&recipients, 0);
+            let current_shares = pool_u64_iterable::shares(&distribution_pool, recipient);
+            let amount_to_distribute = pool_u64_iterable::redeem_shares(
+                &mut distribution_pool,
+                recipient,
+                current_shares
+            );
+
+            // no need to assert on minimum required stake as any next operation of delegator enforces it
+            buy_in_pending_inactive_shares(&mut pool, recipient, amount_to_distribute);
+        };
+        // destroy distribution pool of 0 remaining coins, any dust will be distributed as pending_inactive rewards
+        pool_u64_iterable::destroy_empty(distribution_pool);
+
+        // store `DelegationPool` on the resource account storing the `StakePool` of the destroyed staking contract
+        // `synchronize_delegation_pool` could not have been called as it acquires `DelegationPool` firstly stored here
+        move_to(&stake_pool_signer, pool);
+
+        // store ownership capability of delegation pool on staker
+        move_to(staker, DelegationPoolOwnership { pool_address });
+
+        // enable partial governance voting
+        if (features::partial_governance_voting_enabled(
```

Comment:
I think we can safely skip these feature flags because these have been live for a long time